modules/autocmd.nix

```diff
@@ -0,0 +1,113 @@
+{ config, lib, ... }:
+with lib;
+let
+  helpers = import ../plugins/helpers.nix { inherit lib; };
+
+  autoCmdOption = types.submodule {
+    options = {
+      event = mkOption {
+        type = types.oneOf [
+          types.str
+          (types.listOf types.str)
+        ];
+        description = "The event or events to register this autocommand.";
+      };
+
+      group = mkOption {
+        type = types.nullOr (types.oneOf [
+          types.str
+          types.int
+        ]);
+        description = "The autocommand group name or id to match against.";
+        default = null;
+      };
+
+      pattern = mkOption {
+        type = types.nullOr (types.oneOf [
+          types.str
+          (types.listOf types.str)
+        ]);
+        description = "Pattern or patterns to match literally against.";
+        default = null;
+      };
+
+      buffer = mkOption {
+        type = types.nullOr types.int;
+        description = "Buffer number for buffer local autocommands |autocmd-buflocal|. Cannot be used with <pattern>.";
+        default = null;
+      };
+
+      description = mkOption {
+        type = types.nullOr types.str;
+        description = "A textual description of this autocommand.";
+        default = null;
+      };
+
+      callback = mkOption {
+        type = types.nullOr types.str;
+        description = "The name of a Vimscript function to call when this autocommand is triggered. Cannot be used with <command>}.";
```

Comment:
is this closing bracket meant to be here? 😅

Reply:
Probably not haha